### ISSUE TITLE
Inconsistent contract bytecode when using same fuels-rs version

### ISSUE THREAD

**Issue by simonr0204** [FuelLabs/fuels-rs#122]
Observed : contracts not always compiling to the same bytecode, even when using the same version of fuels-rs.

The resulting contract IDs are therefore different. As an aside, they are also different to the ID obtained when running `forc deploy` (bug reported by @nfurfaro ) , so I assume the bytecode is again different there. 

The bug does not always occur - only on certain contracts. In this case, it seems to depend on the use of the `assert` function from the std-lib. 

Steps to reproduce: 

`git clone git@github.com:simonr0204/contract_id_test.git && cd contract_id_test`
`forc test` 

The test will recompile the same bytecode several times, until the resulting bytecode is inconsistent with the first compilation. Sometimes this happens immediately, sometimes it happens on the the 4th-5th compile. 

In this case, the bytecode ends up being one of two values. In larger contracts, many more bytecode variations are possible.

**Comment by adlerjohn** [comment]
What version of `forc` and `fuels-rs` are you using?

**Comment by simonr0204** [comment]
0.4 for both

**Comment by mohammadfawaz** [comment]
Do you mind quickly testing this with `forc` v0.5.0?

Using `forc deploy`, I keep getting the exact same bytecode/ID for the simple contract you mentioned.

**Comment by simonr0204** [comment]
Just tried with 0.5.0. Same result. Note that I always get the same contract ID when using `forc deploy`. It's when I compile using the SDK :

` let compiled = Contract::compile_sway_contract("./", salt).unwrap();`

that the ID is inconsistent. i.e. different from `forc deploy`, _and_ different when compiling multiple times.

**Comment by mohammadfawaz** [comment]
Thanks Simon. I am still unable to reproduce the issue unfortunately. `forc test` goes on forever and doesn't panic. I'm using the following dependencies:
```
[dependencies]
fuel-gql-client = { version = "0.2", default-features = false }
fuel-tx = "0.3"
fuels-abigen-macro = "0.3"
fuels-contract = "0.3"
fuels-core = "0.3"
rand = "0.8"
tokio = { version = "1.12", features = ["rt", "macros"] }
```
and with both `forc 0.4` and `forc 0.5`. I also tried on Mac and Ubuntu. 

With older versions of `fuels-*`, I kept running into the `missing field `author`` issue.

**Comment by simonr0204** [comment]
This appears resolved when upgrading to `fuels-* = "0.5"`
